src/Domain/Insights/InsightCollectionFactory.php

```diff
@@ -59,17 +59,17 @@ public function get(
         array $metrics,
         OutputInterface $consoleOutput
     ): InsightCollection {
-        $dir = $this->config->getDirectory();
+        $directories = $this->config->getDirectories();
 
         try {
             $files = array_map(static function (\SplFileInfo $file) {
                 return $file->getRealPath();
-            }, $this->filesRepository->within($dir, $this->config->getExcludes())->getFiles());
+            }, $this->filesRepository->within($directories, $this->config->getExcludes())->getFiles());
         } catch (\InvalidArgumentException $exception) {
             throw new DirectoryNotFound($exception->getMessage(), 0, $exception);
         }
 
-        $collector = $this->analyser->analyse($dir, $files);
+        $collector = $this->analyser->analyse($directories[0], $files);
```

Comment:
No need to analyse for each directory ? 

What about launch inspection on `src` and `tests` folders ? The analyse will be it just on `src` ?

Reply:
We don't need directory path for analyser, because it just need to know what files should be analyse. We should pass some files path to analyser and it do the job. Directory path was mainly for shortening path in `Collector`. So basically I can even remove the $directories[0] from analyse function. But this first index in a particular situation is useful (when user only want to analyse one file/dir). So I let it be.

So, if we launch inspection on `src` and `tests` folders, list of both directory files will be send to analyser and everything will be OK.

Reply:
If you have any idea for further improvement, i'm up for that :+1: